src/ekat/logging/ekat_logger.hpp

```diff
@@ -50,36 +50,57 @@ namespace Log = spdlog;
   of its functionality.
 */
 template <typename LogFilePolicy = LogNoFile,    // see ekat_log_file.hpp
-          typename LogMpiPolicy = LogIgnoreRank> // see ekat_log_mpi.hpp
+          typename LogMpiPolicy = LogOnlyRank0> // see ekat_log_mpi.hpp
 class Logger : public spdlog::logger {
   public:
-    Logger() = delete;
 
-    Logger(const std::string& log_name, const std::string& level_str="debug") :
-     spdlog::logger(LogMpiPolicy::name_with_rank(log_name))
+  // primary constructor
+  Logger(const std::string& log_name, const Log::level::level_enum lev=Log::level::debug,
+    const ekat::Comm& mpicomm=ekat::Comm()) :
+    spdlog::logger(LogMpiPolicy::name_with_rank(log_name, mpicomm))
   {
-
-    const auto clevel = Log::level::from_str(level_str);
     auto csink = std::make_shared<spdlog::sinks::stdout_color_sink_mt>();
-    csink->set_level(clevel);
+    csink->set_level(lev);
 
-    std::string fname = LogMpiPolicy::name_with_rank(log_name) + "_logfile.txt";
+    std::string fname = LogMpiPolicy::name_with_rank(log_name, mpicomm) + "_logfile.txt";
     auto fsink = LogFilePolicy::get_file_sink(fname);
 
     this->sinks().push_back(csink);
     this->sinks().push_back(fsink);
 
-    LogMpiPolicy::update_sinks(this->sinks());
+    LogMpiPolicy::update_sinks(this->sinks(), mpicomm);
```

Comment:
See comment above: we should detach the concept of MPI policy from that of C vs F sinks. The Logger should be aware of C vs F, but the policy should only deal with a generic sink. So we should do
```
LogMpiPolicy::update_sink(get_console_sink(), mpicomm);
```